articles/setting-up-cicd-for-python-packages-using-github-actions/index.md

```diff
@@ -0,0 +1,279 @@
+# Setting up CI/CD for Python Packages using GitHub Actions
+
+## Outline
+
+- Prerequisites
+- What is CI/CD
+- Building a Python Package
+- Authenticating GitHub with Test PyPI
+- Packaging & Deploying with GitHub Actions
+- Testing Python Package Locally
+- Conclusion
+
+## Prerequisites
+
+- Understanding of git and GitHub
+- Creating a GitHub repository
+- Creating a Test PyPI account
+- Understanding how to [Build Python Packages](https://www.section.io/engineering-education/building-a-python-package-and-publishing-on-pypi/)
+
+## What is CI/CD
+
+CI/CD comprises three phrases: continuous integration, continuous delivery, and continuous deployment. These phrases seem similar but are of different meanings and implementations. Yet, they are very important to the software development life-cycle.
+
+### Continuous Integration
+
+Continuous Integration is a set of practices that enable development teams to integrate code into version control repositories regularly. According to Atlass*i*an, *It’s a primary DevOps best practice, allowing developers to frequently merge code changes into a central repository where builds and tests then run.*
+
+### Continuous Delivery
+
+Continuous delivery is a software development practice where release to production is done automatically after a code changes.
+
+### Continuous Deployment
+
+Continuous Deployment (CD) is a software release process that uses automated testing to validate if changes to a codebase are correct and stable for immediate autonomous deployment to a production environment.
+
+This [Amazon](https://aws.amazon.com/devops/continuous-delivery/) blog explains in details the difference between these three.
+
+## Building a Python Package
+
+We will attempt to build a basic Python package that tells a user the time in another time zone and works on the command line. The code for this project is in [this repository](https://github.com/edeediong/timezone_checker).
+
+### Application Logic
+
+To build our basic python package, we need two components, namely a `[setup.py](http://setup.py)` file and an `src` folder containing our package logic. Let's start with the `src` folder: it should contain three files, namely `__init__.py`, `[logic.py](http://logic.py)` and `main.py`. 
+
+Copy and paste the following code snippet into `logic.py`.
```

Comment:
Before you start with building the python package, it would be great, if you could mention about what are you going to build (in 2 - 3 lines).

Also, you could add this in a section `Objective`